torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -297,6 +335,10 @@ EntryType ProcessGroupGloo::construct(const AlgorithmKey& key) {
   auto& srcSizes = key.srcSizes;
   entry->src.resize(srcSizes.size());
   for (int i = 0; i < srcSizes.size(); i++) {
+    std::unique_ptr<AutoGPU> gpu;
```

Comment:
Why a `unique_ptr` and an extra allocation? It has a default constructor that doesn't do anything, better to use that.

Reply:
This is now fixed.